### PR TITLE
Add offline migrations to YaST.rb / Card 4616

### PR DESCRIPTION
**Pull request by lagartoflojo**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@tmuntan1 will review your pull request :dancers: check https://trello.com/c/MTT0BvSi/4616-8-sles15-migration-support-online-and-offline-migration-targets
  @tmuntan1: Please review this pull request using our guidelines:
  * test for acceptance criteria / functionality
  * check if the new code is covered with tests
  * check for unintended consequences
  * encourage usage of the boyscout rule
  * make sure the code is architected in the best way
  * check that no unnecessary technical debt got introduced
  * make sure that no unnecessary FIXMEs or TODOs got added

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/72663/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 8f030e0261ea231f148f7da450b0f02de566801f into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

File: lib/suse/connect/api.rb [hunk 1]
```diff
@@ -205,12 +205,21 @@ def system_activations(auth)
       # @param auth [String] authorization string which will be injected in 'Authorization' header in request.
       #   In this case we expect Base64 encoded string with login and password
       # @param [Array <Remote::Product>] a list of producs
+      # @param target_base_product [Remote::Product] (optional) a target base
```
Comment (wstephenson) [inline lib/suse/connect/api.rb]: Needs to specify that this is only used in the :offline case

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/72666/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 40bdfcc0e08d6ba769170794d7087782a9b04a19 into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

File: lib/suse/connect/api.rb (hunk 1 above)
Comment (lagartoflojo) [reply lib/suse/connect/api.rb]: Added!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/72669/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 0510c5af449ad4518ebb6c0a18e7826a53cc8d2f into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/72670/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 386500288dbc26deb155c6a02f730f254b4f1827 into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/72673/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged dd93f71aee879a24ca2e9d22dc6ec14d5aea3b0d into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

File: lib/suse/connect/api.rb [hunk 2]
```diff
@@ -205,12 +205,21 @@ def system_activations(auth)
...
+      #   product to upgrade to. Only used by the backend when kind is :offline. Defaults to nil.
+      # @param kind [Symbol] (optional) :online or :offline. It specifies whether
+      #   the online or the offline migrations are desired. Defaults to :online.
       #
       # @return [Array <Array <Hash>>] the list of possible upgrade paths for the given products,
       #   where each product is represented by a hash with identifier, version, arch and release_type
-      def system_migrations(auth, products)
+      def system_migrations(auth, products, kind: :online, target_base_product: nil)
```
Comment (lagartoflojo) [inline lib/suse/connect/api.rb]: @wstephenson I made the `kind` parameter optional here, but mandatory in [client.rb](https://github.com/SUSE/connect/pull/322/files#diff-ce4a8c41d114461bdba2a49aef6b34a2L146). What do you think, should we keep things consistent and make it mandatory here as well?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/73520/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 049f81c8cb8a9a503df49fc8faf9a6c912bf9f19 into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/73521/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 53e849c3194a8cea2b52ce2c1a051637df4cd52b into f3b1ba7ad3c706679c25897fa919abff67f17f27 )

**Pull request merged**